lib/fog/azurerm/storage.rb

```diff
@@ -32,20 +46,36 @@ def initialize(_options = {})
           end
         end
       end
+
       # This class provides the actual implemention for service calls.
       class Real
         def initialize(options)
           begin
             require 'azure_mgmt_storage'
             require 'azure/storage'
+            @debug = ENV["DEBUG"] || options[:debug]
+            require "azure/core/http/debug_filter" if @debug
           rescue LoadError => e
             retry if require('rubygems')
             raise e.message
           end
 
           credentials = Fog::Credentials::AzureRM.get_credentials(options[:tenant_id], options[:client_id], options[:client_secret])
-          @storage_mgmt_client = ::Azure::ARM::Storage::StorageManagementClient.new(credentials)
-          @storage_mgmt_client.subscription_id = options[:subscription_id]
+          unless credentials.nil?
+            @storage_mgmt_client = ::Azure::ARM::Storage::StorageManagementClient.new(credentials)
+            @storage_mgmt_client.subscription_id = options[:subscription_id]
+          end
+
+          if Fog::Credentials::AzureRM.new_account_credential? options
```

Comment:
Yeah. It make sense. Its fine to make the blob client object in the storage service.